internal/receiver/smartagentreceiver/config.go

```diff
@@ -110,11 +110,18 @@ func (cfg *Config) Unmarshal(componentParser *config.Map) error {
 		return fmt.Errorf("failed setting Smart Agent Monitor config defaults: %w", err)
 	}
 
-	err = setHostAndPortViaEndpoint(cfg.Endpoint, monitorConfig)
+	acceptsEndpoints, err := monitorAcceptsEndpoints(monitorConfig)
 	if err != nil {
 		return err
 	}
 
+	if acceptsEndpoints {
+		err = setHostAndPortViaEndpoint(cfg.Endpoint, monitorConfig)
```

Comment:
following up from slack, we probably want to log that the endpoint is being ignored to prevent unnecessary debugging/confusion.